Untitled Pull Request

dreamWillAppear opened the pull request:
ПР с задачей по 4 спринту

dreamWillAppear committed:
Выполнена задача по 4 спринту

dreamWillAppear commented:
Выполнял задачу в локальном проекте, по завершению подключил репозиторий, переименовал main ветку в sprint_04 и запушил в удаленный репозиторий. Однако добавить ПР никак не удавалось с ошибкой "There are no differences between the source and target branches for the specified pull request."
Удалось решить только созданием чистой ветки и переносом туда файлов из локального проекта.

Helvetii951 commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -4,69 +4,156 @@ final class MovieQuizViewController: UIViewController {
     // MARK: - Lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
+        imageView.layer.cornerRadius = 20
+        show(quiz: convert(model: questions[currentQuestionIndex]))
     }
+    // MARK: - Аутлеты счетчика, обложки фильма, текста вопроса
+    @IBOutlet private var counterLabel: UILabel!
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet private var textLabel: UILabel!
+    
+    private var currentQuestionIndex = 0
+    private var correctAnswers = 0
+    
```
Код стоит организовывать следующим образом:

Публичные переменные
IBOutlet
Приватные переменные
Публичные методы
Приватные методы

Подробнее почитать можно [тут](https://github.com/RedMadRobot/RMR-swift-style-guide#code-organization)

dreamWillAppear replied:
@Helvetii951, привет! 
Большое спасибо за пояснение - как раз не мог понять как это все лучше организовать.

Helvetii951 commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -4,69 +4,156 @@ final class MovieQuizViewController: UIViewController {
+    private var correctAnswers = 0
+    
+    // вью модель для состояния "Вопрос показан"
+  private struct QuizStepViewModel {
+        // картинка с афишей фильма с типом UIImage
+        let image: UIImage
+        // вопрос о рейтинге квиза
+        let question: String
```
Тут чуть поехали отступы. Лучше следить за этим, чтобы код выглядел аккуратнее.

dreamWillAppear committed:
Исправления по работе над 4 спринтом.
- Файлы шрифтов  включил в таргет проекта;

- С целью избежать лишних нажатий пользователем на кнопки и нелегетимных изменений значений счетчика результата - добавил метод disableButtons(_ mustdisabled: Bool), который вызывается в методе показа результата вопроса для отключения кнопок и в методе показа вопроса для активации кнопок вновь.
Таким образом кнопки "Да" и "Нет"  выключаются сразу после нажатия и включаются после после отображения свежих данных;

- Постарался структурировать код следуя рекомендациям;

dreamWillAppear committed:
Сделал аутлеты yesButton и noButton приватными
